### PR TITLE
Implemented a redirect on Staff profile pages.

### PR DESCRIPTION
**Pull request by sasankaj**
This new service looks for the url in the 'External Profile Link' field on staff profile page and redirects the user to the url when the profile page is visited. If the field is empty, the user will not be redirected.

**Note: The External Profile Link'  field is currently empty for all staff profiles on stage. Please add an external URL to one of the profiles before testing.**

### CODE REVIEW

File: web/modules/custom/icfi_utility/src/EventSubscriber/IcfiRedirectStaffPageSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\icfi_utility\EventSubscriber\MyModuleRedirectSubscriber
+ */
+
+namespace Drupal\icfi_utility\EventSubscriber;
+
+use Drupal\Core\Url;
```
Comment (shashank-naik) [inline web/modules/custom/icfi_utility/src/EventSubscriber/IcfiRedirectStaffPageSubscriber.php:10]: I do not think this is needed

File: web/modules/custom/icfi_utility/src/EventSubscriber/IcfiRedirectStaffPageSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+class IcfiRedirectStaffPageSubscriber implements EventSubscriberInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    return([
+      KernelEvents::REQUEST => [
+        ['redirectMyContentTypeNode'],
```
Comment (shashank-naik) [inline web/modules/custom/icfi_utility/src/EventSubscriber/IcfiRedirectStaffPageSubscriber.php:25]: Can we please change this to something like "redirectStaffNode"?

File: web/modules/custom/icfi_utility/src/EventSubscriber/IcfiRedirectStaffPageSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      return;
+    }
+
+    // Redirect only for staff content type.
+    if ($request->attributes->get('node')->getType() !== 'staff') {
+      return;
+    }
+
+    // Get the destination url from the staff profile page.
+    if( $this->getRedirectUrl()) {
```
Comment (shashank-naik) [inline web/modules/custom/icfi_utility/src/EventSubscriber/IcfiRedirectStaffPageSubscriber.php:52]: Minor whitespace issue here

File: web/modules/custom/icfi_utility/src/EventSubscriber/IcfiRedirectStaffPageSubscriber.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+  /**
+   * {@inheritdoc}
+   */
+  private function getRedirectUrl() {
+    $node = \Drupal::routeMatch()->getParameter('node');
+    if (is_string($node)) {
+      $node = Node::load($node);
+    }
+
```
Comment (shashank-naik) [inline web/modules/custom/icfi_utility/src/EventSubscriber/IcfiRedirectStaffPageSubscriber.php:67]: Can we please initialize `$redirect_url = FALSE`. This way the method will either return a URL or FALSE if its not found.

**Comment by sasankaj** [comment]
@shashank-naik, I updated the merge request with the suggested changes. Please review and approve. Thanks!

**Pull request merged**